A66-otel-stats.md

```diff
@@ -0,0 +1,569 @@
+  // The most common way to use this API is -
+  //
+  // OpenTelemetryPluginBuilder().SetMeterProvider(provider)
+  //    .BuildAndRegisterGlobal();
+  //
+  // The set of instruments available are -
+  // grpc.client.attempt.started
+  // grpc.client.attempt.duration
+  // grpc.client.attempt.sent_total_compressed_message_size
+  // grpc.client.attempt.rcvd_total_compressed_message_size
+  // grpc.server.call.started
+  // grpc.server.call.duration
+  // grpc.server.call.sent_total_compressed_message_size
+  // grpc.server.call.rcvd_total_compressed_message_size
+  void BuildAndRegisterGlobal();
+};
+
+```
+
+In the future, additional API might be provided to allow registering the plugin
+for a particular channel or server builder.
+
+#### Java
+
+```java
+public static class OpenTelemetryModuleBuilder {
+    /**
+     * OpenTelemetry instance is used to configure metrics settings.
+     *
+     * Sample
+     *    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+     *         .registerMetricReader(
+     *             PeriodicMetricReader.builder(
+     *                OtlpGrpcMetricExporter.builder().build()).build())
+     *         .build();
+     *
+     *     OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+     *         .setMeterProvider(sdkMeterProvider)
+     *         .build();
+     *
+     * If MeterProvider is not configured, no-op meterProvider will be used by default.
+     * It provides meters which do not record or emit.
+     */
+    public OpenTelemetryModuleBuilder openTelemetry(OpenTelemetry openTelemetry);
+
+    /* If targetFilter returns true for a target, target is recorded as is.
+     * Otherwise it will be recorded as "other". */
+    public OpenTelemetryBuilder targetFilter(Predicate<String> targetFilter);
+
+    public OpenTelemetryModule build();
+}
+```
+
+Note: For non-generated methods, method names are recorded as "other" for
+`grpc.method` attribute. If you are interested in recording the method names for
+these methods, set
+[`isSampledToLocalTracing`](https://grpc.github.io/grpc-java/javadoc/io/grpc/MethodDescriptor.html#isSampledToLocalTracing\(\))
+to `true` while defining your methods in
+[`HandlerRegistry`](https://grpc.github.io/grpc-java/javadoc/io/grpc/HandlerRegistry.html).
+
+#### Go
+
+```go
+import (
+  "go.opentelemetry.io/otel/attribute"
+  "go.opentelemetry.io/otel/metric"
+)
+
+
+package opentelemetry
+
+// MetricsOptions are the metrics options for OpenTelemetry instrumentation.
+type MetricsOptions struct {
+  // MeterProvider is the MeterProvider instance that will be used for access
+  // to Named Meter instances to instrument an application. To enable metrics
+  // collection, set a meter provider. If unset, no metrics will be recorded.
+  // Any implementation knobs (i.e. views, bounds) set in the passed in object
+  // take precedence over the API calls from the interface in this component
+  // (i.e. it will create default views for unset views).
```

Comment:
Nit: this precedence logic is only mentioned in this Go docstring. Can you please mention this somewhere in top level API documentation or a section earlier in the document?

Reply:
This statement should actually be removed since we no longer API to override that

Reply:
Note that there is a statement up above that says - `Users of the gRPC OpenTelemetry plugin will use the OTel SDK's MeterProvider to
[control the views](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view)
and customize the metrics that will be exported.`

Reply:
What? I still call out into API. I thought we needed to since these metrics are unconditionally recorded? That docstring doesn't mention API.

Reply:
Yes, you use the OTel API, but there is no longer any knob to control which instrument gets created and recorded on.  I think the statement "Any implementation knobs (i.e. views, bounds) set in the passed in object take precedence over the API calls from the interface in this component (i.e. it will create default views for unset views)." is just going to confuse people. The term "API" is pretty loaded in this context, and if we want to explain this we should probably mention the whole thing as stated above in the `Language-Specific Details` section.

Reply:
If we want to mention it, instead of mentioning precedence, we should use the word "customize", same as the OTel documentation at https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view. Let me know if you want to update the text here.